arch/arm64/core/CMakeLists.txt

```diff
@@ -38,6 +38,8 @@ zephyr_library_sources_ifdef(CONFIG_HAS_ARM_SMCCC smccc-call.S)
 zephyr_library_sources_ifdef(CONFIG_AARCH64_IMAGE_HEADER header.S)
 zephyr_library_sources_ifdef(CONFIG_SEMIHOST semihost.c)
 zephyr_library_sources_ifdef(CONFIG_DEBUG_COREDUMP coredump.c)
+zephyr_library_sources_ifdef(CONFIG_ARM64_ENABLE_EL3_SERVICE el3_service.S)
+zephyr_library_sources_ifdef(CONFIG_ARM64_ENABLE_EL3_SERVICE el3_service.c)
```

Comment:
I think you can use one single line for this.

Reply:
Sure, I will merge this into one line or actually the whole .S file might go away per your request to move the assembly logic to z_arm64_sync_exc.